src/components/KSelect/KSelectItems.vue

```diff
@@ -59,7 +63,87 @@ const emit = defineEmits<{
   (e: 'selected', item: SelectItem): void
 }>()
 
-const handleItemSelect = (item: SelectItem) => emit('selected', item)
+const handleItemSelect = (item: SelectItem) => {
+  return emit('selected', item)
+}
+
+// Ref to store focusable items
+const focusableItems = ref<HTMLElement[]>([])
+
+// Function to handle the keydown event
+const handleKeyDown = (event: KeyboardEvent, item: SelectItem) => {
+
+  const activeElement = document.activeElement as HTMLElement
+
+  const elements = focusableItems.value
+    .map((el: any) => el.$el || el)
+    .filter((el: HTMLElement) => {
+    // Check if the element has a disabled button among its children
+      const hasDisabledButton = Array.from(el.querySelectorAll('button')).some(button => button.disabled)
+      // Exclude the element if it contains a disabled button
+      return !hasDisabledButton
+    })
+
+  const currentIndex = elements.indexOf(activeElement)
+
+  if (['ArrowUp', 'ArrowDown', 'Tab', ' ', 'Enter'].includes(event.key)) {
```

Comment:
question: shouldn't we be using `code` here instead of `key`?

Reply:
We can, wasnt sure that we have a preference.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code

> The KeyboardEvent.code property represents a physical key on the keyboard (as opposed to the character generated by pressing the key). In other words, this property returns a value that isn't altered by keyboard layout or the state of the modifier keys.

Reply:
Let me check something

Reply:
Ignore this for now